azavea/iow-boundary-tool
# Add reference image opacity slider

## Overview

This PR adds a opacity slider to the sidebar for reference images. It also hooks up the ReferenceImageLayer component to read and set opacity values correctly.

Closes #189 

### Demo

https://user-images.githubusercontent.com/8356789/201738793-68acd58e-21c8-4538-b976-169ebe9a835f.mov

### Notes

Opacity UI updates happen with the slider's `onChangeEnd` event (rather than `onChange`). Using `onChange` is more responsive and works fine on my laptop, but shows significant CPU usage. If we want the UI to be a little bit more responsive, but not as CPU intensive, we could debounce the `onChange` event.

## Testing Instructions

- Edit a boundary
- Add a reference image
- Adjust the opacity slider
  - [x] Ensure the opacity changes with the slider
- Reload the page
  - [x] Ensure the opacity reloads correctly
- Move the reference image and wait for the PUT save request to send
- Reload the window
  - [x] Ensure the opacity reloads correctly

## Checklist
